sdks/python/apache_beam/transforms/core.py

```diff
@@ -3683,8 +3683,14 @@ def _extract_input_pvalues(self, pvalueish):
     return pvalueish, pvalueish
 
   def expand(self, pcolls):
+    windowing = self.get_windowing(pcolls)
     for pcoll in pcolls:
       self._check_pcollection(pcoll)
+      if pcoll.windowing != windowing:
```

Comment:
Shall we just log the error since raising the error could break some users' existing jobs?

Reply:
Yeah, I think that's a good idea. It turns out our test asserts actually do this already for some weird reasons, so I'm sure others do as well. That's what is causing CI to fail - https://github.com/apache/beam/blob/20675c860f46f3f4abce061a6b490166ca68df0f/sdks/python/apache_beam/testing/util.py#L286C7-L286C22